gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -150,129 +154,98 @@ public void setText (BitmapFont font, CharSequence str, int start, int end, Colo
 			if (runEnd != -1) {
 				runEnded:
 				if (runEnd != runStart) { // Can occur eg when a color tag is at text start or a line is "\n".
-					// Store the run that has ended.
-					GlyphRun run = glyphRunPool.obtain();
-					run.color.set(color);
-					fontData.getGlyphs(run, str, runStart, runEnd, lastGlyph);
-					if (run.glyphs.size == 0) {
-						glyphRunPool.free(run);
+					// Store the newRun that has ended.
+					GlyphRun newRun = glyphRunPool.obtain();
+					fontData.getGlyphs(newRun, str, runStart, runEnd, lastGlyph);
+					newRun.y = y;
+					final int rgba8888Color = Color.rgba8888(color);
+					for (int i = 0, n = runEnd - runStart; i < n; i++)
+						newRun.colors.add(rgba8888Color);
+					color = nextColor;
+					if (newRun.glyphs.size == 0) {
+						glyphRunPool.free(newRun);
 						break runEnded;
 					}
-					int xAdvancesCount = run.xAdvances.size;
-					float[] xAdvances = run.xAdvances.items;
-					GlyphRun previous = null;
-					if (lastGlyph != null) { // Move back the width of the last glyph from the previous run.
-						previous = runs.peek();
-						x += xAdvances[0] - previous.xAdvances.peek();
-						xAdvances[0] = 0;
-					}
-					run.x = x;
-					run.y = y;
-					runs.add(run);
 
-					if (newline || runEnd == end) {
-						adjustLastGlyph(fontData, run);
+					if (newline || isLastRun) {
+						adjustXadvanceOfLastGlyph(fontData, newRun);
```

Comment:
The descriptiveness of this private method seems a bit excessive. :)

Reply:
During refactoring I  was often going back to `adjustLastGlyph` to see what exactly is changed so I renamed it to `adjustXadvanceOfLastGlyph` so I knew exactly what it does and what it changes. We can revert the naming of course.